core/src/main/java/org/frankframework/pipes/EscapePipe.java

```diff
@@ -72,42 +72,71 @@ public PipeRunResult doPipe(Message message, PipeLineSession session) throws Pip
 			throw new PipeRunException(this, "cannot open stream", e);
 		}
 
-		String substring = null;
-		String result = input;
-		int i = -1;
-		int j = -1;
-		log.debug("substringStart [{}] substringEnd [{}] input [{}]", substringStart, substringEnd, input);
-		if (substringStart != null && substringEnd != null) {
-			i = input.indexOf(substringStart);
-			if (i != -1) {
-				j = input.indexOf(substringEnd, i);
-				if (j != -1) {
-					substring = input.substring(i + substringStart.length(), j);
-					substring = handle(substring);
-					result = input.substring(0, i + substringStart.length()) + substring + input.substring(j);
-				}
-			}
-		} else {
-			result = handle(input);
+		String result = handleSubstrings(input);
+		return new PipeRunResult(getSuccessForward(), result);
+	}
+
+	private String handleSubstrings(String input) {
+		if (substringStart == null || substringEnd == null) {
+			return handle(input);
 		}
 
-		return new PipeRunResult(getSuccessForward(), result);
+		if (substringStart.isEmpty() || substringEnd.isEmpty()) {
```

Comment:
The `configure()` method already checks for one of these to be `null` but not the other.

I think it is not a sensible configuration to have one set but the other set to empty string.

So instead of doing this check here and returning input unchanged, I think it is better to extend the checks done in `configure()`. Instead of checking for `null`, better to check for `StringUtils.isBlank()` both in `configure()` and in the checks now on line 80. And then remove this case that doesn't change input at all which seems wrong.